### PR TITLE
#37 - Authentication support in proxy

### LINKED ISSUE #37: Upgrade HTTP client and add support for authenticated proxy
When auth support is added to HTTP client we may add authenticated access in Files Proxy

### PR DESCRIPTION
**Pull request by olegmoz**
Closes #37 
Authentication support in proxy

### CODE REVIEW

File: src/test/java/com/artipie/files/FileProxySliceAuthIT.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        this.server.stop();
+    }
+
+    @Test
+    void shouldGet() {
+        MatcherAssert.assertThat(
+            this.proxy.response(
+                new RequestLine(RqMethod.GET, "/foo/bar").toString(), Headers.EMPTY, Content.EMPTY
+            ),
+            new RsHasStatus(RsStatus.OK)
```
Comment (olenagerasimova) [inline src/test/java/com/artipie/files/FileProxySliceAuthIT.java:116]: @olegmoz I'd suggest using `SliceHasResponse` here and also verify response body

**Review by olenagerasimova — CHANGES_REQUESTED** [review]
@olegmoz please check one suggestion

File: src/test/java/com/artipie/files/FileProxySliceAuthIT.java (hunk 1 above)
Comment (olegmoz) [reply src/test/java/com/artipie/files/FileProxySliceAuthIT.java:116]: @olenagerasimova thanks, just tried it and failed - it is incompatible with HTTP client slices. Created an issue about it: https://github.com/artipie/http/issues/251

**Review by olenagerasimova — APPROVED** [review]
@olegmoz thanks

**Comment by olegmoz** [comment]
@olenagerasimova fixed the build, the fail was due to deprecated classes in newer HTTP release. Please re-review

**Review by olenagerasimova — APPROVED** [review]
@olegmoz thanks, all good

**Comment by 0crat** [comment]
Job `gh:artipie/files-adapter#39` is not assigned, can't get performer

**Pull request merged**